src/http.js

```diff
@@ -1,37 +1,83 @@
 /*
     ***** BEGIN LICENSE BLOCK *****
-    
+
     Copyright © 2011 Center for History and New Media
                      George Mason University, Fairfax, Virginia, USA
                      http://zotero.org
-    
+
     This file is part of Zotero.
-    
+
     Zotero is free software: you can redistribute it and/or modify
     it under the terms of the GNU Affero General Public License as published by
     the Free Software Foundation, either version 3 of the License, or
     (at your option) any later version.
-    
+
     Zotero is distributed in the hope that it will be useful,
     but WITHOUT ANY WARRANTY; without even the implied warranty of
     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
     GNU Affero General Public License for more details.
-    
+
     You should have received a copy of the GNU Affero General Public License
     along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
-    
+
     ***** END LICENSE BLOCK *****
 */
 
 var config = require('config');
 var request = require('request');
+var throttledRequestsForURLs = [];
+var startedAt = Date.now();
 var iconv = require('iconv-lite');
 var url = require('url');
 var jsdom = require('jsdom');
 var { JSDOM } = jsdom;
 var wgxpath = require('wicked-good-xpath');
 var MIMEType = require("whatwg-mimetype"); // Use the same MIME type library as JSDOM
 
+
+var domainTimeouts = [
```

Comment:
What happens in a multi-threaded scenario?

I guess this variable should be stored in a some kind of shared memory with access guarded by a mutex, a semaphore or something similar.

Reply:
... maybe i am wrong => if this is just the configuration without write-access, throttledRequestsForURLs might be the one that needs to be guarded.

Reply:
Suppose that even if there were concurrent accesses the threads would insert identical data with the current setup.